File: src/main/java/me/elsiff/morefish/hookers/VaultHooker.java
package me.elsiff.morefish.hookers;

import me.elsiff.morefish.MoreFish;
import net.milkbowl.vault.economy.Economy;
import org.bukkit.plugin.RegisteredServiceProvider;

public class VaultHooker {
    private final MoreFish plugin;
    private Economy econ = null;

    public VaultHooker(MoreFish plugin) {
        this.plugin = plugin;
    }

    public boolean setupEconomy() {
        if (plugin.getServer().getPluginManager().getPlugin("Vault") == null) {
            return false;
        }

        RegisteredServiceProvider<Economy> rsp = plugin.getServer().getServicesManager().getRegistration(Economy.class);
        if (rsp == null) {
            return false;
        }

        econ = rsp.getProvider();
        return econ != null;
    }

    public Economy getEconomy() {
        return econ;
    }
}

File: src/main/kotlin/me/elsiff/morefish/hooker/VaultHooker.kt
package me.elsiff.morefish.hooker

import me.elsiff.morefish.MoreFish
import net.milkbowl.vault.economy.Economy

/**
 * Created by elsiff on 2019-01-04.
 */
class VaultHooker : PluginHooker {
    override val pluginName = "Vault"
    override var hasHooked = false
    lateinit var economy: Economy

    override fun hook(plugin: MoreFish) {
        PluginHooker.checkEnabled(this, plugin.server.pluginManager)

        val registration = plugin.server.servicesManager.getRegistration(Economy::class.java)
        if (registration != null) {
            economy = registration.provider
        }
        hasHooked = true
    }

    fun hasEconomy(): Boolean {
        return ::economy.isInitialized
    }
}